internal/keystore/vault/vault.go

```diff
@@ -67,11 +78,18 @@ func Connect(ctx context.Context, c *Config) (*Store, error) {
 	if c.APIVersion != APIv1 && c.APIVersion != APIv2 {
 		return nil, fmt.Errorf("vault: invalid engine API version '%s'", c.APIVersion)
 	}
-	if (c.AppRole.ID == "" || c.AppRole.Secret == "") && (c.K8S.JWT == "" || c.K8S.Role == "") {
-		return nil, errors.New("vault: no authentication method specified")
+	if c.AppRole != nil && c.K8S != nil {
+		if (c.AppRole.ID == "" || c.AppRole.Secret == "") && (c.K8S.JWT == "" || c.K8S.Role == "") {
+			return nil, errors.New("vault: no authentication method specified")
+		}
+		if (c.AppRole.ID != "" || c.AppRole.Secret != "") && (c.K8S.JWT != "" || c.K8S.Role != "") {
+			return nil, errors.New("vault: ambigious authentication: approle and kubernetes method specified")
```

Comment:
%s/approle and kubernetes method specified/approle and kubernetes both methods specified/g